core/esmf-aspect-meta-model-java/src/main/java/org/eclipse/esmf/aspectmodel/loader/AspectModelLoader.java

```diff
@@ -173,6 +178,37 @@ public AspectModel load( final InputStream inputStream ) {
       return buildAspectModel( loaderContext.loadedFiles() );
    }
 
+   public AspectModel loadFromArchive( final String pathToArchive ) {
+      File zipFile = new File( pathToArchive );
+
+      if ( !zipFile.exists() || !zipFile.isFile() ) {
+         throw new RuntimeException( new FileNotFoundException( "The specified file does not exist or is not a file." ) );
+      }
+
+      List<AspectModelFile> aspectModelFiles = new ArrayList<>();
+
+      try ( ZipFile zip = new ZipFile( zipFile ) ) {
```

Comment:
If I understand correctly, this implementation does not support the structure as descrbied in the ADR-0009 where model roots can be either in the "aspect-models" directory, or in the "aspect-models" subdirectory of any directory in the ZIP's root, or directly in the ZIP's root.
We should either add this in the implementation or create a follow-up issue for supporting those setups.